pkg/dtclient/activegate_query.go

```diff
@@ -65,26 +64,27 @@ func (dtc *dynatraceClient) QueryOutdatedActiveGates(query *ActiveGateQuery) ([]
 	return dtc.QueryActiveGates(query)
 }
 
-func buildQueryParams(query *ActiveGateQuery, logger logr.Logger) string {
-	params := ""
-	if query.Hostname != "" {
-		params += "hostname=" + query.Hostname + "&"
-	}
-	if query.NetworkZone != "" {
-		params += "networkZone=" + query.NetworkZone + "&"
-	}
-	if query.NetworkAddress != "" {
-		params += "networkAddress=" + query.NetworkAddress + "&"
-	}
-	if query.UpdateStatus != "" {
-		params += "updateStatus=" + query.UpdateStatus + "&"
+func (query *ActiveGateQuery) buildQueryParams() string {
+	values := url.Values{}
+	if query != nil {
```

Comment:
Can it still be nil? (I mean, you can create a nil ActiveGateQuery instance and call its method, but in practice is harder.)

Reply:
Regarding the update service:

It would be possible to create a service A that calls `FindOutdatedPods` with a `reconciler` that uses a service B
It doesn't happen in code, but I do like the option that this can be done.
If it causes more problems than it solves I have nothing against removing that, i.e., to let the `updateService` call itself

Reply:
Regarding activegate query:

The `dynatraceClient.QueryActiveGates` call takes a pointer to the query, and handing it nil would result in only the default parameters being used, which imo is expected. 
Otherwise a nil check has to either be introduced in that call or everytime before calling `dynatraceClient.QueryActiveGates`